spec/system/admin/order_cycles/edit_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'system_helper'
+
+RSpec.describe '
+    As an administrator
+    I want to edit a specific order cycle
+' do
+  include AdminHelper
+  include AuthenticationHelper
+  include WebHelper
+
+  context 'when cycle has attached order(s)' do
+    let(:order) { create(:order_without_full_payment) }
+
+    it "show warning modal when datetime field values change" do
+      # When I go to the admin order cycle edit page
+      login_as_admin
+      visit edit_admin_order_cycle_path(order.order_cycle)
+
+      # change non-date range field
+      fill_in 'order_cycle_name', with: "Order cycle name updated"
+      expect(page).to have_content('You have unsaved changes')
+      click_button('Save')
+      expect(page).not_to have_content "Orders are linked to this order cycle"
+      expect(page).to have_content('Your order cycle has been updated.')
+
+      # change date range field value
+      time = DateTime.current
+      find('#order_cycle_orders_close_at').click
+      select_datetime_from_datepicker Time.zone.at(time)
+
+      expect(page.find('#order_cycle_orders_close_at').value).to eq time.strftime('%Y-%m-%d %H:%M')
```

Comment:
This spec doesn't specify what the close_at time _was_ or is changed _to_. 
Apparently they are different in this context, but it's better to make that explicit, for example:
* Create an order cycle with times specified (eg `create(:simple_order_cycle, orders_close_at: `"2024-08-08 00:00"`)
* Create the order for that cycle.
* Specify a time for the selection: `select_datetime_from_datepicker Time.zone.parse("2024-08-09 00:00")`
* Repeat the time for the expect: `expect(page.find('#order_cycle_orders_close_at').value).to eq "2024-08-09 00:00"`

Having the date/times written in the code makes it easier to read and understand the spec, which saves time and helps us avoid mistakes.